### PR TITLE
feat: support points write method

### PR DESCRIPTION
**Pull request by Chenxulin97**

### CODE REVIEW

File: opengemini/write.go [hunk 1]
```diff
@@ -56,3 +56,33 @@ func (c *client) WriteBatchPoints(database string, bp *BatchPoints) error {
...
+	var writer io.Writer
+
+	if c.config.GzipEnabled {
+		writer = gzip.NewWriter(&buffer)
+	} else {
+		writer = &buffer
+	}
+
+	if _, err := io.WriteString(writer, point.String()); err != nil {
+		callbackFunc(nil, err)
```
Comment (goyjy) [inline opengemini/write.go:72]: if error, need return?

File: opengemini/write.go [hunk 2]
```diff
@@ -56,3 +56,33 @@ func (c *client) WriteBatchPoints(database string, bp *BatchPoints) error {
...
+	}
+
+	if closer, ok := writer.(io.Closer); ok {
+		if err := closer.Close(); err != nil {
+			callbackFunc(nil, err)
```
Comment (goyjy) [inline opengemini/write.go:77]: like up

File: opengemini/client.go [hunk 3]
```diff
@@ -18,7 +19,9 @@ type Client interface {
 	Ping(idx int) error
 	Query(query Query) (*QueryResult, error)
 
-	// WriteBatchPoints batch points to assigned database
+	// WritePoint write single point to assigned database
+	WritePoint(database string, point *Point, callbackFunc func(*http.Response, error))
```
Comment (xuthus5) [inline opengemini/client.go:23]: `WritePoint` method signature contains an `error` return.

File: opengemini/write.go [hunk 4]
```diff
@@ -56,3 +56,46 @@ func (c *client) WriteBatchPoints(database string, bp *BatchPoints) error {
...
+
+	var writer io.Writer
+
+	if c.config.GzipEnabled {
+		writer = gzip.NewWriter(&buffer)
+	} else {
+		writer = &buffer
+	}
+
+	if _, err := io.WriteString(writer, point.String()); err != nil {
```
Comment (shoothzj) [inline opengemini/write.go:71]: should we check if the return length exactly equals point serialize length?

File: opengemini/write.go [hunk 5]
```diff
@@ -56,3 +56,46 @@ func (c *client) WriteBatchPoints(database string, bp *BatchPoints) error {
...
+		return err
+	}
+
+	if closer, ok := writer.(io.Closer); ok {
```
Comment (shoothzj) [inline opengemini/write.go:75]: why we must cast to `io.Closer` type?

Comment (Chenxulin97) [reply opengemini/write.go:75]: GzipWriter’s Close method closes itslef by flushing any unwritten data to the underlying io.Writer and writing the GZIP footer.It does not close the underlying io.Writer.

File: opengemini/write.go (hunk 4 above)
Comment (Chenxulin97) [reply opengemini/write.go:71]: io.Writer and gzip.Writer both operate bytes.Buffer ,  WriteString appends the contents of s to the buffer, growing the buffer as needed. The return value n is the length of s; err is always nil. If the buffer becomes too large, WriteString will panic with ErrTooLarge.

**Review by shoothzj — APPROVED** [review]

**Pull request merged**